core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala

```diff
@@ -137,8 +163,8 @@ protected[controller] class RestAPIVersion_v1(
             sendCorsHeaders {
                 (pathEndOrSingleSlash & get) {
                     complete(OK, info)
-                } ~ authenticate(basicauth) {
-                    user =>
+                } ~ authenticate(basicauth) { user =>
+                    checkPutSize(config, transid) {
```

Comment:
this is going to be tied into entitlement at some point.
it may be worth moving this into the routes that actually support PUT and avoiding the check on ALL API requests.